pay-api/src/pay_api/services/statement.py

```diff
@@ -471,6 +399,15 @@ def populate_overdue_from_invoices(statements: List[StatementModel]):
             statement.is_overdue = overdue_statements.get(statement.id, 0) > 0
         return statements
 
+    @staticmethod
+    def get_payment_methods_from_details(invoice_detail_tuple, pay_account) -> str:
+        """Grab payment methods from invoices detail tuple."""
+        payment_methods = {payment_method for _, payment_method, _,
+                           auth_account_id in invoice_detail_tuple if pay_account.auth_account_id == auth_account_id}
+        if not payment_methods:
+            payment_methods = {pay_account.payment_method}
```

Comment:
Will this work for old empty statements that is not the current payment method? 

For example, if I am currently EFT and I view an empty statement from a few months ago that is another payment method.

Reply:
Yes that's exactly the case this covers

Reply:
Should be fixed take a look again